cmd/tailscale/cli/up.go

```diff
@@ -221,9 +224,17 @@ func runUp(ctx context.Context, args []string) error {
 			if n.ErrMessage != nil {
 				fatalf("backend error: %v\n", *n.ErrMessage)
 			}
-			if s := n.State; s != nil {
+			if upArgs.forceReauth && !loggedin {
+				loggedin = true
+				printed = true
+				bc.StartLoginInteractive()
```

Comment:
Two lines up you set `loggedIn = true`, so why do you need to start logged in here? Probably that other variable could be named better.